tests/components/hyperion/test_light.py

```diff
@@ -0,0 +1,430 @@
+        DOMAIN, SERVICE_TURN_ON, {ATTR_ENTITY_ID: TEST_ENTITY_ID}, blocking=True
+    )
+
+    assert not client.async_send_clear.called
+    assert not client.async_send_set_effect.called
+
+
+async def test_light_async_turn_off(hass):
+    """Test turning the light off."""
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
+
+    client.async_send_set_component = CoroutineMock(return_value=True)
+    await hass.services.async_call(
+        DOMAIN, SERVICE_TURN_OFF, {ATTR_ENTITY_ID: TEST_ENTITY_ID}, blocking=True
+    )
+
+    assert client.async_send_set_component.call_args == call(
+        **{
+            const.KEY_COMPONENTSTATE: {
+                const.KEY_COMPONENT: const.KEY_COMPONENTID_LEDDEVICE,
+                const.KEY_STATE: False,
+            }
+        }
+    )
+
+    # No calls if no state loaded.
+    client.has_loaded_state = False
+    client.async_send_set_component = CoroutineMock(return_value=True)
+    call_registered_callback(client, "client-update", {"loaded-state": False})
+
+    await hass.services.async_call(
+        DOMAIN, SERVICE_TURN_OFF, {ATTR_ENTITY_ID: TEST_ENTITY_ID}, blocking=True
+    )
+
+    assert not client.async_send_set_component.called
+
+
+async def test_light_async_updates_from_hyperion_client(hass):
+    """Test receiving a variety of Hyperion client callbacks."""
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
+
+    # Bright change gets accepted.
+    brightness = 10
+    client.adjustment = [{const.KEY_BRIGHTNESS: brightness}]
+    call_registered_callback(client, "adjustment-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["brightness"] == round(255 * (brightness / 100.0))
+
+    # Broken brightness value is ignored.
+    bad_brightness = -200
+    client.adjustment = [{const.KEY_BRIGHTNESS: bad_brightness}]
+    call_registered_callback(client, "adjustment-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["brightness"] == round(255 * (brightness / 100.0))
+
+    # Update components.
+    client.is_on.return_value = True
+    call_registered_callback(client, "components-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.state == "on"
+
+    client.is_on.return_value = False
+    call_registered_callback(client, "components-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.state == "off"
+
+    # Update priorities (V4L)
+    client.is_on.return_value = True
+    client.visible_priority = {const.KEY_COMPONENTID: const.KEY_COMPONENTID_V4L}
+    call_registered_callback(client, "priorities-update")
+    entity_state = hass.states.get(TEST_ENTITY_ID)
+    assert entity_state.attributes["icon"] == hyperion_light.ICON_EXTERNAL_SOURCE
+    assert entity_state.attributes["hs_color"] == (0.0, 0.0)
+    assert entity_state.attributes["effect"] == const.KEY_COMPONENTID_V4L
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6233107)